Tests/EventDispatcherTests.swift

```diff
@@ -0,0 +1,198 @@
+                    let anotherDispatcher = EventDispatcher()
+
+                    eventDispatcher.listenToOnce(anotherDispatcher, eventName: eventName, callback: callback)
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == true
+
+                    callbackWasCalled = false
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                }
+            }
+
+            describe("off") {
+                it("doesn't executes callback function if it is removed") {
+                    let listenId = eventDispatcher.on(eventName, callback: callback)
+                    eventDispatcher.off(listenId)
+                    eventDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                }
+                it("doesn't execute callback if it is removed, but the others are called") {
+                    var anotherCallbackWasCalled = false
+                    let anotherCallback: EventCallback = { _ in
+                        anotherCallbackWasCalled = true
+                    }
+
+                    let listenId = eventDispatcher.on(eventName, callback: callback)
+                    eventDispatcher.on(eventName, callback: anotherCallback)
+
+                    eventDispatcher.off(listenId)
+                    eventDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                    expect(anotherCallbackWasCalled) == true
+                }
+            }
+
+            describe("stopListening") {
+                it("disables all callbacks") {
+                    eventDispatcher.on(eventName, callback: callback)
+                    eventDispatcher.on("another-event", callback: callback)
+
+                    eventDispatcher.stopListening()
+
+                    eventDispatcher.trigger(eventName)
+                    eventDispatcher.trigger("another-event")
+
+                    expect(callbackWasCalled) == false
+                }
+
+                it("cancels all callback functions on only one context object") {
+                    let anotherDispatcher = EventDispatcher()
+                    var anotherCallbackWasCalled = false
+                    anotherDispatcher.on(eventName) { _ in
+                        anotherCallbackWasCalled = true
+                    }
+
+                    eventDispatcher.on(eventName, callback: callback)
+
+                    eventDispatcher.stopListening()
+
+                    eventDispatcher.trigger(eventName)
+                    anotherDispatcher.trigger(eventName)
+
+                    expect(callbackWasCalled) == false
+                    expect(anotherCallbackWasCalled) == true
+                }
+
+                it("cancels a specific callback function for an event on a given context object") {
```

Comment:
This ``it`` seems that is holding a context logic, do you think that is necessary to split into a ``context`` > ``it`` ?
It could be:
```swift
describe("#stopListening") {
    context("When calls stopListening for a event") {
        it("cancels a specific callback") {}
    }
}
```